payment/internal/server/authorize_card.go

```diff
@@ -0,0 +1,71 @@
+package server
+
+import (
+	"context"
+	"time"
+
+	"github.com/GrabItYourself/giys-backend/lib/authutils"
+	"github.com/GrabItYourself/giys-backend/lib/postgres/models"
+	"github.com/GrabItYourself/giys-backend/payment/internal/libproto"
+	"github.com/omise/omise-go"
+	"github.com/omise/omise-go/operations"
+)
+
+func (s *Server) AuthorizeCard(ctx context.Context, in *libproto.AuthorizeCardRequest) (*libproto.AuthorizeCardResponse, error) {
+	userId, _, err := authutils.ExtractUserFromGrpcContext(ctx)
+	if err != nil {
+		return nil, err
```

Comment:
don't just return `err`.
Return `status.Error(c codes.Code, msg string)` instead to indicate correct grpc status code.
See https://pkg.go.dev/google.golang.org/grpc/codes@v1.49.0#Code

Reply:
fixed